### ISSUE TITLE
Wiki

### ISSUE THREAD

**Issue by Utumno** [wrye-bash/wrye-bash#67]
Does not support categories I'm afraid - went ahead and renamed the articles to group better (was taking time to spot the article one was interested in and would only get worse). I am afraid now I might have broken existing links - this says something - no links from this wiki (except if we agree on some pages). This is our dev wiki.

I open this to discuss wiki related things:

- Agree on the [dev], [git] , [github] etc ? Also z will be used to archive pages as in [z git svn]
- Keep notes on the wiki, pages that need editing etc. 

For instance what is the bashed patches one (edit there) ?

Posts that we need to write:

- [ ] Recommended editors for python - you can just write about the one you use here
- [ ] Pycharm guide
- [ ] rebase workflows (see #151, #156)

Posts that we need to revisit:

- [x] Issues and labels (https://github.com/wrye-bash/wrye-bash/wiki/%5Bgithub%5D-Branching-Model-&-Using-The-Repository) - also add to the main readme. `ready` -> review, ~~in progress~~
- [ ] https://github.com/wrye-bash/wrye-bash/wiki/%5Bgithub%5D-Branching-Model-&-Using-The-Repository --> fix- feature- --> issue-

General:

- [ ] Wrap lines. This way it is easy to check diffs in the wiki. Existing posts will be a pain to wrap - will look into some automated tool - but it might break the markdown...

**Comment by Utumno** [comment]
@lojack5: Is this still valid (from the [[github] Reporting a bug](https://github.com/wrye-bash/wrye-bash/wiki/%5Bgithub%5D-Reporting-a-bug) page:

* If you're using WBSA, things are a bit trickier. There are currently a few known issues:
  * 1. Wrye Bash wont work correctly if you launch it from the command line (DOS). Always launch by double clicking the exe or a shortcut to it.
  * 2. Due to issue #1, no debug mode is available. Some errors that happen will still be printed either to a "Wrye Bash.exe.txt" file, or to the "stdout/stderr" window that shows up with errors sometimes.

Seems that Wrye Bash.exe -d from a cmd works fine. Also is there a Wrye Bash.exe.txt anymore ?
Is the current state as described in 946f203b4f8be09b51ea9c80781d88e49fb5623d ?

I really need this wiki page set up correctly for release (and then copy pasted to the forums thread starts, nexus etc) - please :)

**Comment by lojack5** [comment]
Not sure at this point anymore.  I know some code has changed in bash.py, that sets up stdout/stderr if the `-d` command line option is detected, so I *think* that covers all the issues that used to exist, but I'm not sure.

I haven't tested, but beyond what I stated above, the real issue with WBSA running with `-d` wasn't that `deprints` wouldn't get printed, but that they'd go to the wrong spot.  When running in Python mode, all `deprints` go to the log file, and never show up in the wxPython stdout/stderr window.  WBSA couldn't properly wrest control of stdout/sterr from wxPython, so as soon as wxPython took over, the debug log would stop being filled with useful information, basically negating the purpose of launching with `-d`: to generate a log file.

The whole reason *that* was happening was because launching via the command line instead of double clicking had issues (which I can't recall), meaning we couldn't use the piping operator to pipe the output to the debug log (which is how Python mode accomplishes that).

TL;DR: not sure.  You could do some tests:
1. Put some test `deprints` in place, probably one after [this](https://github.com/wrye-bash/wrye-bash/blob/8627c425d93e4078f41ad104c5216e56cc9b6ad3/Mopy/bash/bash.py#L288) line which initializes stdout/stderr.  And some more, maybe throw it in some menu option or something.  Somewhere that's easy for you to trigger it once the main GUI is running.

2. Try launching WBSA via the command line instead of shortcuts/double clicking, but by typing `"-`Wrye Bash.exe" -d`.

3. Check to see if the `deprint`s are going to the debug text file, or if they're popping up in GUI (in the wxPython stdout/stderr window).

4. `Wrye Bash.exe.txt`:  Yes, sort of.  In most cases a user should never see this file,  because as noted in the link to bash.py above, we almost immediately redirect stdout/stderr to the `BashBugDump.log` file instead.  However, if somewhere along the lines an accidental `print` or `deprint` slips in before that section  of code happens, then it *will* go to `Wrye Bash.exe.txt`.  Ideally any instances of this should be caught before a release, so end users will never see the file.

**Comment by Utumno** [comment]
Thanks indeed - one last point - what the heck _is_ Wrye Bash.exe.txt ? I mean where is even defined and where is to be created ?

I have already run `Mopy>"Wrye Bash.exe" -d` and the bug dump is pretty fine dumped but will do some more testing as you suggest. What the rest all really means is that the booting sequence (including imports, wx availability etc) must be revisited - I think I already had an issue on that...

Thanks

**Comment by lojack5** [comment]
`Wrye Bash.exe.txt` is something that py2exe does when building a windows exe (not a console exe though).  It sets up stderr (and maybe stdout, not sure on that one) to initially go to a text file with the same name as the executable.  That's inconsistent with the Python version, so we redirect it to the same file name that Python version uses, but of course if anything happens before that, it'll still end up in `Wrye Bash.exe.txt`.

**Comment by lojack5** [comment]
Ah, [here's](http://www.py2exe.org/index.cgi/StderrLog) the info on py2exe, took me a while to find it.

**Comment by Utumno** [comment]
Thanks lo - it would take me double to infinite time :)

So - I will change the instruction to "have an eye open for Wrye Bash.exe.txt" and see if debug prints fine and call this a day

**Comment by Utumno** [comment]
Well long story short what _does not_ run in my configuration is `C:\path\to\Mopy>"Wrye Bash Launcher.pyw" -d`  while `C:\path\to\Mopy>c:\path\to\Python27\python(w).exe "Wrye Bash Launcher.pyw" -d` will work just fine as well as `C:\path\to\Mopy>"Wrye Bash.exe" -d`.

Added the info here:https://github.com/wrye-bash/wrye-bash/wiki/%5Bgithub%5D-Reporting-a-bug

Btw what crashes Bash to desktop on boot (the Supierce issue) is `C:\Dropbox\eclipse_workspaces\python\wrye-bash\Mopy>c:\_\Python27\pythonw.exe "Wrye Bash Launcher.pyw"` when there is a print + a flush before binding the stdout/stderr. `C:\Dropbox\eclipse_workspaces\python\wrye-bash\Mopy>c:\_\Python27\python.exe "Wrye Bash Launcher.pyw"` works just fine

Added the info here: https://github.com/wrye-bash/wrye-bash/wiki/%5Bdev%5D-Gotchas

**Comment by Metallicow** [comment]
@Utumno  concerning your:"""C:\path\to\Mopy>"Wrye Bash Launcher.pyw" -d"""; the only sane result is that the .py or .pyw extension is not correctly registered with windows. See regedit or your sys "PATH" env variable Otherwise you have to provide the "full path" in quotes. (preferably for both)

That or the debug bat file is not setting the correct cwd before executing.

**Comment by lojack5** [comment]
If you're launching with python.exe, it launches in console mode, so you wouldn't have the CTD issue.  If you launch with pythonw.exe, then you would.

As for your trying to launch by just typing the file name - might be an registered extension issue are you saying it fails to even attempt to start, or that it crashes after starting?.  If it' the first, then yeah, sounds like an issue with how the .pyw extension is registered in windows.

**Comment by Utumno** [comment]
Thanks guys - yeah what I meant is that `C:\path\to\Mopy>"Wrye Bash Launcher.pyw" -d` does not produce the bashbugdump.log - that is it 'ignores' the -d argument. The rest of the commands take it in as expected
